TestingHelpers/MockFileSystemWatcher.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.ComponentModel;
+
+namespace System.IO.Abstractions.TestingHelpers
+{
+    public class MockFileSystemWatcher : FileSystemWatcherBase
+    {
+        public override bool IncludeSubdirectories { get; set; }
+        public override bool EnableRaisingEvents { get; set; }
+        public override string Filter { get; set; }
+        public override int InternalBufferSize { get; set; }
+        public override NotifyFilters NotifyFilter { get; set; }
+        public override string Path { get; set; }
+#if NET40
+        public override ISite Site { get; set; }
+        public override ISynchronizeInvoke SynchronizingObject { get; set; }
+#endif
```

Comment:
This is probably mostly for me, but where are you finding that certain members need to be specific to 4.0? I've dug through the documentation for the FileSystemWatcher (https://msdn.microsoft.com/en-us/library/system.io.filesystemwatcher(v=vs.110).aspx) as well as a couple others, but I'm not seeing anything that definitively says to wrap these.

I thought it was going to be the "Version Information" towards the bottom, but then say DriveInfo's GetAccessControl (https://msdn.microsoft.com/en-us/library/yk1cww0c(v=vs.110).aspx) only lists 4.0, but we don't have the NET40 check around that method.

Reply:
To follow up, it looks like it's in the FileSystemWatcherBase, though I'm still not even sure how the original author found that requirement.

Reply:
Yeah I only put those in because of the base class, otherwise it wouldn't compile for both .net framework and .net standard. 

It appears that .net standard doesn't have the `ISite` and `ISynchronizeInvoke` interfaces defined in them (as Microsoft pulled them out into separate nuget packages).  That may or may not have existed when the base class was written and so the `#if` was put in place (at least that's my guess).